File: frontend/src/types/conversation.ts
export interface ConversationCharacter {
  id: string
  displayName: string
  titles: string[]
  species: string
  homeTerritory: string | null
}

export interface ConversationMessage {
  messageId: string
  sender: 'USER' | 'BANDAR' | 'SYSTEM'
  timestamp: string
  content: string
  visibility: string
  metadata: Record<string, string>
}

export interface ConversationResponse {
  conversationId: string
  sessionId: string
  currentCharacter: ConversationCharacter
  startedAt: string
  lastActivity: string
  status: string
}

export interface SendMessageRequest {
  content: string
}

export interface AppendMessagesResponse {
  messages: ConversationMessage[]
}

export interface ConversationMessagesResponse {
  messages: ConversationMessage[]
}

File: backend/src/main/java/com/chugalkhorbandar/domain/conversation/Conversation.java
package com.chugalkhorbandar.domain.conversation;

import java.time.Instant;
import java.util.List;

public record Conversation(
        String conversationId,
        String sessionId,
        ConversationCharacter currentCharacter,
        Instant startedAt,
        Instant lastActivity,
        ConversationStatus status,
        List<ConversationMessage> messages) {

    public Conversation withLastActivity(Instant activityAt) {
        return new Conversation(
                conversationId, sessionId, currentCharacter, startedAt, activityAt, status, messages);
    }

    public Conversation withMessages(List<ConversationMessage> updatedMessages) {
        return new Conversation(
                conversationId, sessionId, currentCharacter, startedAt, lastActivity, status, updatedMessages);
    }

    public Conversation closed() {
        return new Conversation(
                conversationId, sessionId, currentCharacter, startedAt, lastActivity, ConversationStatus.CLOSED, messages);
    }
}
